packages/nextjs/test/serverIntegrations.test.ts

```diff
@@ -0,0 +1,54 @@
+import { RewriteFrames } from '@sentry/integrations';
+import { Integration } from '@sentry/types';
+
+import { defaultRewriteFrames, getFinalServerIntegrations } from '../src/utils/serverIntegrations';
+
+interface IntegrationFunction {
+  (integrations: Integration[]): Integration[];
+}
+
+describe('integrations without RewriteFrames', () => {
+  test('as an array', () => {
+    const integrations: Integration[] = [];
+    // Should get a single integration: RewriteFrames
+    let finalIntegrations = getFinalServerIntegrations(integrations);
+    expect(Array.isArray(finalIntegrations)).toBeTruthy();
+    finalIntegrations = finalIntegrations as Integration[];
+    expect(finalIntegrations.length === 1).toBeTruthy();
+    expect(finalIntegrations[0]).toMatchObject(defaultRewriteFrames);
+  });
+
+  test('as a function', () => {
+    const integrations: IntegrationFunction = (): Integration[] => {
+      return [];
+    };
+    // Should get a single integration: RewriteFrames
+    const integrationWrapper = getFinalServerIntegrations(integrations);
+    expect(typeof integrationWrapper === 'function').toBeTruthy();
+    const finalIntegrations = (integrationWrapper as IntegrationFunction)([]);
+    expect(finalIntegrations.length === 1).toBeTruthy();
+    expect(finalIntegrations[0]).toMatchObject(defaultRewriteFrames);
+  });
+});
+
+describe('integrations with RewriteFrames', () => {
+  test('as an array', () => {
+    const rewriteFramesIntegration = new RewriteFrames();
+    const integrations = [rewriteFramesIntegration];
```

Comment:
As you did in the next test:

```suggestion
     const userIntegrations = [ new RewriteFrames() ];
```